content/log/processing.md

```diff
@@ -0,0 +1,199 @@
+It transforms this log:
+{{< highlight json >}}
+{
+"message":"john clicked on a link : www.datadoghq.com for app dd_app",
+"hostname":"host_1",
+"level":"info"
+}
+{{< /highlight >}}
+
+Into this log:
+{{< highlight json >}}
+{
+"message":"john clicked on a link : www.datadoghq.com for app dd_app",
+"hostname":"host_1",
+"severity":"info"
+}
+{{< /highlight >}}
+
+However, beware that each incoming severity value is mapped as follows:
+
+* If this is an integer from 0 to 7 we map it to the [syslog severity standards](https://en.wikipedia.org/wiki/Syslog#Severity_level)
+* If it starts with **emerg** or **f** (case unsensitive) we map it to **emerg (0)**
+* If it starts with **a** (case unsensitive) we map it to **alert (1)**
+* If it starts with **c** (case unsensitive) we map it to **critical (2)**
+* If it starts with **err** (case unsensitive) we map it to **error (3)**
+* If it starts with **w** (case unsensitive) we map it to **warning (4)**
+* If it starts with **n** (case unsensitive) we map it to **notice (5)**
+* If it starts with **i** (case unsensitive) we map it to **info (6)**
+* If it starts with **d**, trace or verbose (case unsensitive) we map it to **debug (7)**
+* Any other condition, we map it to **info (6)**
+
+### Attribute Remapper
+
+This processor remaps any attribute(s) to another one, 
+for instance here it remaps “user_id” to “user.id”
+
+{{< img src="log/processing/attribute_remapper_processor_tile.png" alt="Attribute remapper processor tile" >}}
+
+It transforms this log:
+{{< highlight json >}}
+{
+"message":"john clicked on a link : www.datadoghq.com for app dd_app",
+"hostname":"host_1",
+"severity":"info",
+"my":{
+    "source":{
+        "attribute":"this is some good attribute value"
+            }
+    }
+}
+{{< /highlight >}}
+Into this log:
+{{< highlight json >}}
+{
+"message":"john clicked on a link : www.datadoghq.com for app dd_app",
+"hostname":"host_1",
+"severity":"info",
+"my":{
+    "target":{
+        "attribute":"this is some good attribute value"
+        }
+    }
+}
+{{< /highlight >}}
+
+### URL processor 
+
+This processor extracts query params and other important parameter from an url, just enter the source attribute of your url:
+{{< img src="log/processing/url_processor.png" alt="Url Processor" >}}
+
+### Useragent parser
+
+UserAgent processor takes a useragent attribute and does its best to extract all the meaningful information the OS, the browser and the device.
+It is also able to recognize the main bots like the Google Bot, Yahoo Slurp, Bing and others.
+
+Some servers or systems (such as IIS) encode useragents, if so ask this processor to **Apply a URL decode first** before parsing it.
+
+This settings: 
+{{< img src="log/processing/useragent_processor_tile.png" alt="Useragent processor tile" >}}
+
+Give the following results:
+{{< img src="log/processing/useragent_processor.png" alt="Useragent processor" >}}
+
+## What's next
+
+* Learn how to explore your logs [here](/log/explore)
```

Comment:
> Learn how to explore your logs [here]

Learn how to [explore your logs](/log/explore)

Reply:
👍